lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -408,8 +408,8 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
         $list = [];
         foreach ($tableForeignKeys as $value) {
             $value = array_change_key_case($value, CASE_LOWER);
-            $name  = $value['constraint_name'];
-            if (! isset($list[$name])) {
+            $index = $value['constraint_name'] ?? $value['id'];
```

Comment:
Why does the list need to have the constraint name as the key? There's still a potential for collision (the identifiers are numeric strings: `0`, `1`, `2`…).